eng/common/scripts/Invoke-DevOpsAPI.ps1

```diff
@@ -2,6 +2,24 @@
 
 $DevOpsAPIBaseURI = "https://dev.azure.com/{0}/{1}/_apis/{2}/{3}?{4}api-version=6.0"
 
+function Get-Base64EncodedToken([string]$AuthToken)
+{
+  $unencodedAuthToken = "nobody:$AuthToken"
+  $unencodedAuthTokenBytes = [System.Text.Encoding]::UTF8.GetBytes($unencodedAuthToken)
+  $encodedAuthToken = [System.Convert]::ToBase64String($unencodedAuthTokenBytes)
+
+  if ($isDevOpsRun) {
+    # We are doing this here so that there is zero chance that this token is emitted in Azure Pipelines
+    # build logs. Azure Pipelines will see this text and register the secret as a value it should *** out
+    # before being transmitted to the server (and shown in logs). It means if the value is accidentally
+    # leaked anywhere else that it won't be visible. The downside is that when the script is executed
```

Comment:
You wrote:

>  It means if the value is accidentally leaked anywhere else that it won't be visible.

So if I am reading it right, ADO works like that:
- somebody tries to output text "foo"
- ADO checks and sees that "foo" was being previously output (which is here) as a secret value, so it added it to its "secret values map", so it stars it out.

?